query-engine/core/src/query_graph_builder/extractors/query_arguments.rs

```diff
@@ -99,6 +100,17 @@ fn process_order_object(
         Some((field_name, field_value)) => {
             let field = model.fields().find_from_all(&field_name)?;
             match field {
+                Field::Relation(rf) if rf.is_list => {
+                    path.push(rf.clone());
+
+                    let object: ParsedInputMap = field_value.try_into()?;
+                    let (sort_aggregation, sort_order) = sort_aggregation_from_order_object(object)?;
+                    let sf: Vec<_> = rf.related_model().primary_identifier().scalar_fields().collect();
+                    // TODO: Return a better error?
+                    let id = sf.first().unwrap();
```

Comment:
We pick the first index field. Is it possible to have a field without unique/index ? If so, what kind of error should I return here?

Reply:
The primary identifier is potentially a combination of fields where single fields are not unique. The weakness in the approach that you picked is that it's not obvious how you would represent `count(*)` (which is what you're aiming for), instead you're picking random fields. I propose:
- you either come up with an abstraction that allows you to easily use field or *
- ... or you ignore the field for count aggregation queries for now and do a `count(*)` always.

A valid abstraction can be in the `SortAggregation`, where you have `Count { all: bool }`.